tests/audit_api_test.go

```diff
@@ -0,0 +1,96 @@
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/phoenixnap/go-sdk-bmc/auditapi"
+	"github.com/stretchr/testify/suite"
+)
+
+type AuditApiTestSuite struct {
+	suite.Suite
+	ctx           context.Context
+	configuration *auditapi.Configuration
+	apiClient     *auditapi.APIClient
+}
+
+func (suite *AuditApiTestSuite) SetupTest() {
+	// Set configuration
+	suite.configuration = auditapi.NewConfiguration()
+	suite.configuration.Host = "127.0.0.1:1080"
+	suite.configuration.Scheme = "http"
+	// Set the context background
+	suite.ctx = context.WithValue(context.Background(), "accessToken", "ACCESSTOKENSTRING")
+	suite.ctx = context.WithValue(suite.ctx, "serverIndex", nil)
+	fmt.Println(">>> From SetupSuite")
+	suite.apiClient = auditapi.NewAPIClient(suite.configuration)
+}
+
+// this function executes after all tests executed
+func (suite *AuditApiTestSuite) TearDownTest() {
+	fmt.Println(">>> From TearDownSuite")
+	TestUtilsImpl{}.resetExpectations()
+}
+
+func verifyCalledOnce(suite *AuditApiTestSuite, expectationId string) {
+	// Result retrieved from server's verification
+	// Verifying expectation matched exactly once.
+	verifyResult := TestUtilsImpl{}.verifyExpectationMatchedTimes(expectationId, 1)
+
+	// Asserts a successful result
+	suite.Equal(202, verifyResult.StatusCode)
+}
+
+func (suite *AuditApiTestSuite) Test_get_events_all_query_params() {
+
+	// Generate payload
+	request, response := TestUtilsImpl{}.generatePayloadsFrom("auditapi/events_get", "./payloads")
+
+	// Extract the response expectation id
+	expectationId := TestUtilsImpl{}.setupExpectation(request, response, 1)
+
+	// Fetch a map of query parameters
+	qpMap := TestUtilsImpl{}.generateQueryParams(request)
+
+	// Converting to int32
+	limitstring, ok := qpMap["limit"].(string)
+	if !ok {
+		panic(ok)
+	}
+
+	from, _ := time.Parse(time.RFC3339, request.QueryStringParameters[0].Values[0]) // time.Time | From the date and time (inclusive) to filter event log records by. (optional)
```

Comment:
Should we keep these comments? I believe they're from the README right?

Reply:
yeah they can be removed